File: Sources/llamaObjCxx/session/LlamaSessionConfig.m
//
//  LlamaSessionConfig.m
//  llama
//
//  Created by Alex Rozanski on 13/03/2023.
//

#import "LlamaSessionConfig.h"

@implementation _LlamaSessionConfig

@synthesize mode = _mode;
@synthesize numberOfThreads = _numberOfThreads;
@synthesize numberOfTokens = _numberOfTokens;
@synthesize reversePrompt = _reversePrompt;
@synthesize seed = _seed;

- (instancetype)initWithMode:(_LlamaSessionMode)mode
{
  if ((self = [super init])) {
    _mode = mode;
  }

  return self;
}

@end

File: Sources/llama/config/LlamaSessionConfig.swift
//
//  LlamaSessionConfig.swift
//  llama
//
//  Created by Alex Rozanski on 29/03/2023.
//

import Foundation
import llamaObjCxx

public final class LlamaSessionConfig: SessionConfig, ObjCxxParamsBuilder {
  public static var `default`: Self {
    return self.mergeIntoDefaults(
      from: Self.init(numTokens: 128, hyperparameters: Hyperparameters())
    )
  }

  func build(for modelURL: URL) -> _LlamaSessionParams {
    return SessionConfigBuilder(sessionConfig: self, mode: .regular).build(for: modelURL)
  }
}
